modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2341,36 +2384,314 @@ public int compare(CodegenProperty one, CodegenProperty another) {
+                }
+                if (hasDiscriminatorCnt == composedSchema.getAnyOf().size() && discriminatorsPropNames.size() == 1) {
+                    disc.setPropertyName(foundDisc.getPropertyName());
+                    disc.setMapping(foundDisc.getMapping());
+                    return disc;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This function is only used for composed schemas which have a discriminator
+     * Process oneOf and anyOf models in a composed schema and adds them into
+     * a list if the oneOf and anyOf models contain
+     * the required discriminator. If they don't contain the required
+     * discriminator or the discriminator is the wrong type then an error is
+     * thrown
+     * @param composedSchemaName The String model name of the composed schema where we are setting the discriminator map
+     * @param discPropName The String that is the discriminator propertyName in the schema
+     * @param c The ComposedSchema that contains the discriminator and oneOf/anyOf schemas
+     * @param openAPI The OpenAPI spec that we are using
+     * @return the list of oneOf and anyOf MappedModel that need to be added to the discriminator map
+     */
+    protected List<MappedModel> getOneOfAnyOfDescendants(String composedSchemaName, String discPropName, ComposedSchema c, OpenAPI openAPI) {
+        ArrayList<List<Schema>> listOLists = new ArrayList<List<Schema>>();
+        listOLists.add(c.getOneOf());
+        listOLists.add(c.getAnyOf());
+        List<MappedModel> descendentSchemas = new ArrayList();
+        for (List<Schema> schemaList: listOLists) {
+            if (schemaList == null) {
+                continue;
+            }
+            for  (Schema sc: schemaList) {
+                String ref = sc.get$ref();
+                if (ref == null) {
+                    // for schemas with no ref, it is not possible to build the discriminator map
+                    // because ref is how we get the model name
+                    // we only hit this use case for a schema with inline composed schemas, and one of those
+                    // schemas also has inline composed schemas
+                    // Note: if it is only inline one level, then the inline model resolver will move it into its own
+                    // schema and make it a $ref schema in the oneOf/anyOf location
+                    throw new RuntimeException("Invalid inline schema defined in oneOf/anyOf in '" + composedSchemaName + "'. Per the OpenApi spec, for this case when a composed schema defines a discriminator, the oneOf/anyOf schemas must use $ref. Change this inline definition to a $ref definition");
+                }
+                CodegenProperty df = discriminatorFound(composedSchemaName, sc, discPropName, openAPI);
+                String modelName = ModelUtils.getSimpleRef(ref);
+                if (df == null || !"string".equals(df.dataType) || df.required != true) {
```

Comment:
Not all languages use "string". What about using `df.isString` to check if it's a string instead?

Same for line 2568.

Reply:
Switched to using df.isString